### DISCUSSION TITLE
Source code in one file

### DISCUSSION THREAD

**Discussion by motgyl** [networkx/networkx#7384] [General]
does anyone have the source code in one file, due to the nature of my project I can't use multiple files

**Comment by rossbar** [comment]
The short answer is: no

NetworkX is a [Python package](https://docs.python.org/3/tutorial/modules.html#packages) and thus is structured in this way. It is not possible to simply dump the source code of the entire package into a single file, at least not without a lot of manual effort to "flatten" the import structure etc.

If you want to use NetworkX code in your own project, you should use the standard Python pattern of importing the package (or the sub-modules/sub-packages you need).
